### ISSUE TITLE
TextField label falls down over the entered value if the TextField is enclosed in an Elevation.

### ISSUE THREAD

**Issue by fairps** [jamesmfriedman/rmwc#143]
### What RMWC Version are you using?
1.1.2

### What browser(s) and React Version is this bug affecting?
react 16.2, 
Chrome Version 63.0.3239.132 (Official Build) (64-bit)

### What build system are you using?
> Create React App

### What are the steps to reproduce the bug?
The .mdc-text-field__label--float-above class get removed from the labels on the TextField if the TextField is contained within an Elevation.

The following JSX will recreate the issue:
```
          <Elevation
            style={{border:'solid 1px lightgray',borderRadius:5}}
            z={this.state.elevation || 0} transition
            onMouseOver={() => this.setState({elevation: 10})}
            onMouseOut={() => this.setState({elevation: 0})}
          >
            <div >
              <label style={{fontFamily:'Kalam',fontSize:'1.5rem',color:'Gray',marginLeft:'1rem',marginTop:'1rem'}}>Welcome back!</label>
                <form style={{margin:'0rem 1rem 1rem 1rem'}}>
                  <div>
                    <TextField label="Email" type="email" />
                  </div>
                  <div>
                    <TextField label="Password" type="password" />
                  </div>

                </form>
                <Button style={{float:'right'}}>Login</Button>
            </div>
          </Elevation>
```
1) Click in the TextFields and enter values
![image](https://user-images.githubusercontent.com/27831128/36067471-9f54f388-0e83-11e8-93b9-b68a37b6ffeb.png)

2) Start clicking back and forth between the fields, or onto the button and the .mdc-text-field__label--float-above class gets lost from the labels causing them to fall down even though there are values present.
![image](https://user-images.githubusercontent.com/27831128/36067520-1fd93f8c-0e84-11e8-9c50-d0c601b2e120.png)

### What is the expected behavior?
Labels should stay above values after they are entered

### What is the actual behavior?
Label is falling down over the value

### Any other information you believe would be useful?
If I remove the Elevation object then everything works perfectly.

**Comment by jamesmfriedman** [comment]
I’ll look into this. My initial reaction is that this is an uncontrolled text input and react is re-rendering it causing it to lose its className. A temporary fix before the bug gets fixed would be to make them controlled components.

**Comment by izakfilmalter** [comment]
I'm having this issue with controlled inputs atm.

**Comment by fairps** [comment]
At least in my case you are correct James... making them controlled did seem to fix the problem.

**Comment by lo-tp** [comment]
Basically this happened because the `TextField` component is used in the uncontrolled way.
Following is a modified version of your code which works pretty well inside a `Elevation` component.
```javascript
class TextFieldExample extends React.Component {
  state = {
    value: ''
  };

 changeEmail = (evt) => {
   this.setState({
     email: evt.target.value });
 }
 changePassword = (evt) => {
   this.setState({
     pswd: evt.target.value });
 }

 render() {
   return (
     <Elevation
       style={{ border: 'solid 1px lightgray', borderRadius: 5 }}
       z={this.state.elevation || 0} transition
       onMouseOver={() => this.setState({ elevation: 10 })}
       onMouseOut={() => this.setState({ elevation: 0 })}
     >
       <div >
         <label style={{ fontFamily: 'Kalam', fontSize: '1.5rem', color: 'Gray', marginLeft: '1rem', marginTop: '1rem' }}>Welcome back!</label>
         <form style={{ margin: '0rem 1rem 1rem 1rem' }}>
           <div>
             <TextField onChange = {this.changeEmail} value={this.state.email} label="Email" type="email" />
           </div>
           <div>
             <TextField onChange = {this.changePassword} value={this.state.pswd}label="Password" type="password" />
           </div>

         </form>
         <Button style={{ float: 'right' }}>Login</Button>
       </div>
     </Elevation>
   );
 }
}
```

**Comment by jamesmfriedman** [comment]
Correct @lo-tp. We'll be making changes to avoid these problems in the future, but for now uncontrolled components can get out of sync with material-web-components.

**Comment by lo-tp** [comment]
@jamesmfriedman 
I am writing a PR to fix this problem.

**Comment by lo-tp** [comment]
Check out @ #145

**Comment by jamesmfriedman** [comment]
Fixed in 1.2.0
